OpenRA.Server/Program.cs

```diff
@@ -30,14 +30,17 @@ static void Main(string[] args)
 
 			// Special case handling of Game.Mod argument: if it matches a real filesystem path
 			// then we use this to override the mod search path, and replace it with the mod id
-			var modArgument = arguments.GetValue("Game.Mod", null);
+			var modID = arguments.GetValue("Game.Mod", null);
 			var explicitModPaths = new string[0];
-			if (modArgument != null && (File.Exists(modArgument) || Directory.Exists(modArgument)))
+			if (modID != null && (File.Exists(modID) || Directory.Exists(modID)))
 			{
-				explicitModPaths = new[] { modArgument };
-				arguments.ReplaceValue("Game.Mod", Path.GetFileNameWithoutExtension(modArgument));
+				explicitModPaths = new[] { modID };
+				modID = Path.GetFileNameWithoutExtension(modID);
 			}
 
+			if (modID == null)
+				throw new ArgumentException("The mod to launch was not specified", "mod");
```

Comment:
How about "Game.Mod argument missing or mod could not be found"? It's a bit clearer, and gives the user the info how to fix the error.